library/certificate_request.py

```diff
@@ -38,6 +38,12 @@
         certificate.
     required: false
     default: certmonger
+  directory:
+    description:
+      - Directory where certificate and key will be stored. Only used
+        if I(name) is not a fully qualified path.
```

Comment:
I would say "absolute path" instead of "fully qualified path"

Reply:
Amended.

Reply:
By the way, what will happen if `name` does not start with a slash, but it contains slash(es)? Will it put the certificate file under a subdirectory of `directory`?

Reply:
Correct. The paths are joined using `os.path.join`.

Reply:
if `name is `foo/bar` and there is no `/etc/pki/tls/certs/foo` directory, will it be created? I don't see it in the source, but maybe that would be certmonger's job.
By the way, in case of subdirectories, should the prefix be `/etc/pki/tls/certs/foo` or `/etc/pki/tls/foo/certs`?
The `certs` and `private` subdirectories are not distribution-specific?

Reply:
@pcahyna good points!

Although `certs` and `private` directories could probably be considered distro specific they are used inside the default path of pretty much everything I found: OpenBSD, Slack, Alpine, Gentoo and deb-based systems.

In case of sub-directories /etc/pki/ should be the correct base directory for rpm-based systems. For all the other cases /etc/ssl should not contain sub-directories.
Because of that I'm thinking that name should be either an absolute path or a simple string and the module should raise an error if a non-abspath is given.

Does that make sense?

Reply:
Is it usual to have user- or package- defined subdirectories under `/etc/pki/`? If so, are they directly under `/etc/pki` or under `/etc/pki/tls`? If they are not expected, it makes indeed the best sense to forbid them.

Reply:
In the case of absolute path, no `certs`/`private` path components are inserted. Is it intended and expected?

Reply:
On RPM-based there are package defined sub-directories under `/etc/pki` (without `tls`).

In that case I'm going to perform the changes to only allow simple string or abspaths.

Reply:
So if anyone needs subdirectories, they should be implemented as a separate option instead.

Reply:
Amended.